### PR TITLE
ci: reenable link checker & fix broken links

### PR DESCRIPTION
**Pull request by ethanndickson**
Follow-up on #15484.

### CODE REVIEW

**Comment by ethanndickson** [comment]
* **#15489** <a href="https://app.graphite.dev/github/pr/coder/coder/15489?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ethanndickson and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/admin/users/idp-sync.md [hunk 1]
```diff
@@ -327,8 +327,7 @@ the OIDC provider. See
 > ones include `groups`, `memberOf`, and `roles`.
 
 Next configure the Coder server to read groups from the claim name with the
-[OIDC organization field](../../reference/cli/server.md#--oidc-organization-field)
-server flag:
+OIDC organization field server flag:
```
Comment (ethanndickson) [inline docs/admin/users/idp-sync.md:330]: This CLI arg got moved to runtime config a few days ago, so it's no longer in the reference. The docs will need updating soon.

File: .github/.linkspector.yml [hunk 2]
```diff
@@ -1,9 +1,19 @@
 dirs:
   - docs
+excludedDirs:
+  # Downstream bug in linkspector means large markdown files fail to parse
+  # but these are autogenerated and shouldn't need checking
```
Comment (ethanndickson) [inline .github/.linkspector.yml:5]: https://github.com/micromark/micromark/blob/dd1bbe0c80fe4c4d1e3ee7adeaff1d8e81e3818f/packages/micromark-util-subtokenize/dev/lib/splice-buffer.js#L121-L123
https://stackoverflow.com/questions/22123769/rangeerror-maximum-call-stack-size-exceeded-why

File: .github/workflows/weekly-docs.yaml [hunk 3]
```diff
@@ -33,6 +33,7 @@ jobs:
           reporter: github-pr-review
           config_file: ".github/.linkspector.yml"
           fail_on_error: "true"
+          filter_mode: "nofilter"
```
Comment (ethanndickson) [inline .github/workflows/weekly-docs.yaml:36]: I incorrectly assumed that the weekly-docs job being started from the cron job would mean it wouldn't use the pull request changes filter, but evidently not. This means it checks the whole `docs` folder for the cron job & on PRs.

**Review by matifali — APPROVED** [review]

File: .github/workflows/weekly-docs.yaml (hunk 3 above)
Comment (matifali) [reply .github/workflows/weekly-docs.yaml:36]: So on schedule, we check the whole repo?
What does the `filter_mode` mean?

Comment (ethanndickson) [reply .github/workflows/weekly-docs.yaml:36]: The default filter_mode was just the PR diff. Now it should be everything mentioned in the `.linkspector.yml` config.

**Pull request merged**